contracts/libs/Ibc.sol

```diff
@@ -245,8 +251,8 @@ library Ibc {
     function parseAckData(bytes calldata ack) external pure returns (AckPacket memory ackData) {
         // this hex value is '"result"'
         ackData = (keccak256(ack[1:9]) == keccak256(hex"22726573756c7422"))
-            ? AckPacket(true, Base64.decode(string(ack[11:ack.length - 2]))) // result success
-            : AckPacket(false, ack[10:ack.length - 2]); // this is an error
+            ? AckPacket(AckStatus.SUCCESS, Base64.decode(string(ack[11:ack.length - 2]))) // result success
+            : AckPacket(AckStatus.FAILURE, ack[10:ack.length - 2]); // this is an error
```

Comment:
See the other comment, but we actually can't change the `AckPacket` return type here, since `parseAckData` is used to find the proof key in the `acknowledge` method in dispatcher. We need to have two seperate ackPacket types now